### ISSUE TITLE
Job not running when expected (every 10 minutes)

### ISSUE THREAD

**Issue by daveomcd** [ondrejbartas/sidekiq-cron#229]
![image](https://user-images.githubusercontent.com/3517432/47452994-8ec45600-d799-11e8-902e-f527e03729b9.png)

I'm attempting to run the job every 10 minutes.  I believe I have the cron set correctly for every 10 minutes (shown in picture).  But the last enqueued shows it doesn't execute every 10 minutes.  

I'm running...

- Rails 5.1.6 
- Sidekiq 5.1.3 (upgraded to 5.2.2 after posting - same results)
- Sidekiq-Cron 1.0.0 (upgraded to 1.0.4 after posting - same results)

**Comment by jmettraux** [comment]
Hello,

So, the job doesn't execute every 10 minutes. It was enqueued once and you took a screenshot 11 minutes after that.

Anything showing up in the logs? Errors, enqueue attempts?
If it doesn't execute every 10 minutes, does it execute at all? Are you seeing no executions or sporadic executions, not every 10 minutes?

Thanks in advance for giving this issue tracker a clearer view on your problem.

**Comment by daveomcd** [comment]
Yes you have that right.  I've checked in the `sidekiq.log` and it shows when I manually kick it off through the UI with the Enqueue  button.
```

2018-10-24T21:28:03.450Z 397 TID-11a7m9 CreateOrUpdateDariExamJob JID-d10c6c5100b89534df431593 INFO: start
2018-10-24T21:28:07.419Z 397 TID-11a7m9 CreateOrUpdateDariExamJob JID-d10c6c5100b89534df431593 INFO: done: 3.969 sec
```
I also see this in the `development.log`

`[ActiveJob] [CreateOrUpdateDariExamJob] [33c53033-9f2c-4a11-8546-6146df213d2b] Performed CreateOrUpdateDariExamJob (Job ID: 33c53033-9f2c-4a11-8546-6146df213d2b) from Sidekiq(import_queue) in 3967.74ms`

However if I wait for 10 minutes it never shows up.  And thanks for your help I'll try/provide anything that will help. Thanks!

**Comment by daveomcd** [comment]
Here are some other files that I felt might be relevant...

**sidekiq.yml**
```
---
:concurrency: 25
:pidfile: ./tmp/pids/sidekiq.pid
:logfile: ./log/sidekiq.log
:queues:
  - default
  - [high_priority, 2]
  - [search_queue, 2]  
  - [import_queue, 3]
  - [user_queue, 4]
  - [small_queue, 5]
  - [rss_parser, 6]
:daemon: true
```

**initializers/sidekiq.rb**

```
schedule_file = "config/schedule.yml"

if File.exist?(schedule_file) #&& Sidekiq.server?
  Sidekiq::Cron::Job.load_from_hash YAML.load_file(schedule_file)
end
```

**Comment by daveomcd** [comment]
Also the "last enqueued" column is now saying "in 3 hours".  Not sure how that factors into this if at all?

**Comment by jmettraux** [comment]
Have you tried with a trivial "hello world" job every minute (`* * * * *`) ?

**Comment by daveomcd** [comment]
I think it all has to do with me misunderstanding my deployment strategy, and how the sidekiq processes operate.  As I had setup sidekiq with capistrano a long time ago, I had forgotten some of the ins-and-outs.

I've removed the comment in my **initializers/sidekiq.rb** file.  So that the "&& Sidekiq.server?" is placed back.  And it has corrected to the time it's being asked to run at (~every 10 minutes).  I had commented it out originally due to this issue #224 because I couldn't add new jobs to my schedule.yml and have it reflected on the /cron web ui.  By commenting out the Sidekiq.server? condition it would add the job when I deployed.
